src/cli.rs

```diff
@@ -0,0 +1,128 @@
+//! Define the gossiping subcommands.
+
+use std::str::FromStr as _;
+
+use anyhow::{Context, Result};
+use clap::{ArgGroup, Subcommand};
+use futures_lite::StreamExt;
+use futures_util::SinkExt;
+use iroh_net::NodeId;
+use tokio::io::AsyncBufReadExt;
+
+use crate::rpc::client::SubscribeOpts;
+
+/// Commands to manage gossiping.
+#[derive(Subcommand, Debug, Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GossipCommands {
+    /// Subscribe to a gossip topic
+    #[command(
+        long_about = r#"Subscribe to a gossip topic
+
+Example usage:
+
+    $ iroh gossip subscribe --topic test --start
```

Comment:
make remove the `iroh` prefix, given this is now not guranteed to be the binary name